asv/benchmarks/atomics.py

```
@@ -166,3 +225,162 @@ def time_cuda(self, vals_np_dict, dtype_str):
+
+    The benchmark compares the normal atomic-add path against deterministic
+    scatter-sort-reduce for the same kernel using CUDA graph replay. A small
+    size sweep exposes where deterministic execution crosses over. Two
+    destination counts are used:
+
+    - ``1``: worst-case contention, where every thread targets the same output.
+    - ``65536``: lower contention, closer to a scatter workload.
+    """
+
+    params = (["normal", "deterministic"], [1, 65536], DETERMINISTIC_BENCHMARK_SIZES)
+    param_names = ["mode", "num_outputs", "num_elements"]
```

Ruff (0.15.9)

[warning] Mutable default value for class attribute

(RUF012)